# HW 1 - Sentiment Analysis for Employee Satisfaction during COVID-19 Pandemic on Glassdoor Panel Dataset

The COVID-19 pandemic and the following lockdowns disrupted company operations and lead to an unprecedented impact on employees. Companies adopting different strategies in dealing with this urgent condition may have different results on employee satisfaction. Employee satisfaction reviews during the pandemic may contain useful information for investors. In this homework, we will use Sentiment Analysis to analyze employee satisfaction reviews on the Glassdoor dataset from major Silicon Valley firms.

**Submission**
Submit one Jupyter notebook file (.ipynb) with your code; be sure to explain your code using Markdown cells and comments.  No need to attach any other file.

In [1]:
autosave 60

Autosaving every 60 seconds


In [2]:
# print all the outputs in a cell
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [3]:
import pandas as pd

## Step 1: Import the datasets
1. data_glassdoor_hw1.csv
2. positive.txt
3. negative.txt
4. stopwords.txt

### Import data_glassdoor_hw1.csv

In [4]:
df = pd.read_csv('data_glassdoor_hw1.csv')

In [5]:
df.head(10)

Firm           date               employee_title           location  \
0  Facebook  Aug 31, 2020           Engineering Manager     Menlo Park, CA   
1  Facebook  Aug 31, 2020             Software Engineer     Menlo Park, CA   
2  Facebook  Aug 31, 2020           Senior Data Analyst     Menlo Park, CA   
3  Facebook  Aug 31, 2020    Critical Facility Engineer     Huntsville, AL   
4  Facebook  Aug 30, 2020                     Anonymous  San Francisco, CA   
5  Facebook  Aug 30, 2020             Software Engineer       San Jose, CA   
6  Facebook  Aug 28, 2020       Privacy Program Manager       New York, NY   
7  Facebook  Aug 28, 2020                      Manager          Austin, TX   
8  Facebook  Aug 28, 2020                      Director       New York, NY   
9  Facebook  Aug 27, 2020            Product Specialist        Seattle, WA   

                              review_title  \
0                Neat place. Org struggles   
1                    Feel lucky to join FB   
2               great, but could be better   
3       Facebook is a great place to work.   
4                          Good management   
5                                 Nice job   
6                Phenomenal place to work!   
7  Good Place to Grow and Employee Focused   
8                        Left but Returned   
9                                Ambiguity   

                                                pros  \
0  Well funded, lots of resources, focus on engin...   
1  - Lovely people must be the first pros.\n- Boo...   
2             great tech, smartest ppl, great food\n   
3  you can be yourself, more diversity than my ot...   
4  Good managers with experience in hand\nLot to ...   
5       Competitive compensation and good benefits\n   
6  Compensation, benefits and work-life balance a...   
7  Solid Growth Trend, Lots of great perks, Compa...   
8  - Best people\n- Challenging / interesting wor...   
9  Lots of documentation available online\nTalent...   

                                                cons  
0  Organization is a challenge, senior people fig...  
1  I don't feel any cons so far\nBe the first to ...  
2               lack of policing political bigotry\n  
3                        no cons to report at all.\n  
4  Too much politics in the company\nBe the first...  
5  a little bit overtime possibly\nBe the first t...  
6  Learning curve can be steep, particularly in t...  
7  Back office tools and processes need some work...  
8                               - None to think of\n  
9  You’re a generalist in a sea of specialists. V...

#### Get 6 lists from the dataframe: 
1. apple_pros 
2. apple_cons 
3. facebook_pros 
4. facebook_cons 
5. google_pros 
6. google_cons

In [6]:
apple_pros=list(df.loc[(df['Firm'] == 'Apple')]['pros'])
apple_cons=list(df.loc[(df['Firm'] == 'Apple')]['cons'])

facebook_pros=list(df.loc[(df['Firm'] == 'Facebook')]['pros'])
facebook_cons=list(df.loc[(df['Firm'] == 'Facebook')]['cons'])

google_pros=list(df.loc[(df['Firm'] == 'Google')]['pros'])
google_cons=list(df.loc[(df['Firm'] == 'Google')]['cons'])

Please print out first 10 reviews for each list

In [7]:
#print first 10 reviews for each list
apple_pros[:11]
apple_cons[:11]

facebook_pros[:11]
facebook_cons[:11]

google_pros[:11]
google_cons[:11]

['fun ppl and exciting work days\n',
 'Easy paced\nCustomer interactions\nTechnology based\nFun to work in\nFrequent breaks\n',
 "Infinite complex work, never-ending series of problems to solve, never bored. Great for the under achiever types, because you'll never actually finish anything.\n",
 'Cool people and decent benefits\n',
 'Best retail company to work for\n',
 "It's very awesome, indeed it is.\n",
 'flexible schedule great starting pay many perks diverse environment\n',
 'Great at everything yeah oh yah\n',
 'It was pretty good time and staff was great.\n',
 'Get to work with a lot of different people with different interests/expertises, healthy work life balance, discounts on stocks & other benefits\n',
 'Highly innovative and constant challenges\n']

['hours for retail sucks fr\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 'Pressure to sell apple care\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 "Sell your soul, it's the American way.\nBe the first to find this review helpful\nHelpful\nShare\nRepor",
 'Never promoted women in the business field\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 'The job is in retail\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 'Long hours are very long indeed.\n',
 'long training process it is full time for two months\nContinue reading\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 'nothing at all really nothing\n',
 'Nothing that I can think of\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 'Expect a day to be kind of draining, it is BUSY. There will be moments your patience is tested by both customers & other employees,\nBe the first to find this review helpful\nHelpful\n

['Well funded, lots of resources, focus on engineering.\n',
 '- Lovely people must be the first pros.\n- Bootcamp is a really good buffer for new grad.\n- Flexible team selection\n- Big tolerant with making mistakes\n',
 'great tech, smartest ppl, great food\n',
 'you can be yourself, more diversity than my others jobs, I am supported by upper leadership.\n',
 'Good managers with experience in hand\nLot to learn\n',
 'Competitive compensation and good benefits\n',
 'Compensation, benefits and work-life balance are great! There are plenty of opportunities to work on meaningful and impactful projects. Management has been super supportive with the transition to remote work and working from home due to COVID.\n',
 'Solid Growth Trend, Lots of great perks, Company genuinely tries to do the right thing\n',
 '- Best people\n- Challenging / interesting work\n- A culture that really cares about their people\n- There are very few companies that truly let their employees build the future. While o

['Organization is a challenge, senior people fighting for the the best reviews (which trigger significant payouts). Rigorous performance management process which has many flaws (politics, anecdotal discussions, selective review collection).\n',
 "I don't feel any cons so far\nBe the first to find this review helpful\nHelpful\nShare\nRepor",
 'lack of policing political bigotry\n',
 'no cons to report at all.\n',
 'Too much politics in the company\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 'a little bit overtime possibly\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 'Learning curve can be steep, particularly in the first few months.\n2 people found this review helpful\nHelpful\nShare\nRepor',
 'Back office tools and processes need some work, Some entitlement in the workforce\n',
 '- None to think of\n',
 'You’re a generalist in a sea of specialists. Very hard to get buy-in for your role.\n2 people found this review helpful\nHelpful\nShare\n

['- Highly competitive pay, best benefits I\'ve seen, best perks I\'ve seen (yes, this doesn\'t matter, but when everything else is good, this helps).\n- Very strong individuals. The average engineer here is very skilled and some of the best in the industry are here. This does not mean everyone is a genius, but there are certainly some here. To do the work, you definitely don\'t need to be the best, but the quality here is definitely among the best I\'ve seen (I\'ve worked at other "top" companies).\n- Stellar work life balance. I\'ve seen reviews for this that are negative, but I feel like they\'re on the wrong team or something. My team was busier than most and promoted people quickly and I still rarely pushed over 40 hours a week. I\'ve seen some people work significantly fewer hours as well and still progressed. You can coast at L4 if you just do basic work and rest and vest.\n- Good projects if you\'re on the right team. If you get lucky and get on the right team then the work is 

['- Work does not translate to other jobs. You never get to work on stuff like Apache Spark, AWS S3, RabbitMQ, and other technologies that are very common in industry. Instead you learn things like BigTable, which is similar to Apache HBase but not exactly. Your knowledge of internal frameworks is mostly useless in other jobs. The concepts learned are not, however.\n- Half the projects are objectively boring and stale. Most of the projects are maintaining some aspect of the system (cloud, ads, etc) and a lot of it is tedious infrastructure work.\n- Office politics as usual. The general population is friendly, but this does not mean everyone. Office politics definitely do play a role.\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 'None, love what you do, do what you love\nBe the first to find this review helpful\nHelpful\nShare\nRepor',
 "Haven't found one Servant Leader. Finance Leadership talks the talks but doesn't walk the walk. Constantly put pressure on middl

### Import Positive words 

Save import words in a list

Hint: Files I/O

In [8]:
#open positive.txt file as read only
f = open('positive.txt', 'r')

In [9]:
# make a blank list for positive words
pos_list = []

In [10]:
#make a for loop to make the pro word list
for word in f:
    word_strip = word.strip()
    pos_list.append(word_strip)

In [11]:
f.close()

In [12]:
#print the pos_list to check
print(pos_list)

['a+', 'abound', 'abounds', 'abundance', 'abundant', 'accessable', 'accessible', 'acclaim', 'acclaimed', 'acclamation', 'accolade', 'accolades', 'accommodative', 'accomodative', 'accomplish', 'accomplished', 'accomplishment', 'accomplishments', 'accurate', 'accurately', 'achievable', 'achievement', 'achievements', 'achievible', 'acumen', 'adaptable', 'adaptive', 'adequate', 'adjustable', 'admirable', 'admirably', 'admiration', 'admire', 'admirer', 'admiring', 'admiringly', 'adorable', 'adore', 'adored', 'adorer', 'adoring', 'adoringly', 'adroit', 'adroitly', 'adulate', 'adulation', 'adulatory', 'advanced', 'advantage', 'advantageous', 'advantageously', 'advantages', 'adventuresome', 'adventurous', 'advocate', 'advocated', 'advocates', 'affability', 'affable', 'affably', 'affectation', 'affection', 'affectionate', 'affinity', 'affirm', 'affirmation', 'affirmative', 'affluence', 'affluent', 'afford', 'affordable', 'affordably', 'afordable', 'agile', 'agilely', 'agility', 'agreeable', 'ag

### Import Negative words

Save import words in a list

Hint: Files I/O

In [13]:
# open file
f = open('negative.txt', 'r')

In [14]:
# make a blank list
neg_list = []

In [15]:
# make a for loop to make the negative word list
for word in f:
    word_strip = word.strip()
    neg_list.append(word_strip)

In [16]:
# close the file
f.close()

In [17]:
# print the negative list to check
print(neg_list)

['2-faced', '2-faces', 'abnormal', 'abolish', 'abominable', 'abominably', 'abominate', 'abomination', 'abort', 'aborted', 'aborts', 'abrade', 'abrasive', 'abrupt', 'abruptly', 'abscond', 'absence', 'absent-minded', 'absentee', 'absurd', 'absurdity', 'absurdly', 'absurdness', 'abuse', 'abused', 'abuses', 'abusive', 'abysmal', 'abysmally', 'abyss', 'accidental', 'accost', 'accursed', 'accusation', 'accusations', 'accuse', 'accuses', 'accusing', 'accusingly', 'acerbate', 'acerbic', 'acerbically', 'ache', 'ached', 'aches', 'achey', 'aching', 'acrid', 'acridly', 'acridness', 'acrimonious', 'acrimoniously', 'acrimony', 'adamant', 'adamantly', 'addict', 'addicted', 'addicting', 'addicts', 'admonish', 'admonisher', 'admonishingly', 'admonishment', 'admonition', 'adulterate', 'adulterated', 'adulteration', 'adulterier', 'adversarial', 'adversary', 'adverse', 'adversity', 'afflict', 'affliction', 'afflictive', 'affront', 'afraid', 'aggravate', 'aggravating', 'aggravation', 'aggression', 'aggress

### Import Stopwords

Save import words in a list

Hint: Files I/O

In [18]:
# open file
f = open('stopwords.txt', 'r')

In [19]:
#make a blank list
stop_list = []

In [20]:
#make a for loop to make the stop word list
for word in f:
    word_strip = word.strip()
    stop_list.append(word_strip)

In [21]:
#close the stop words file
f.close()

In [22]:
#print the stop word list to check
print(stop_list)

['a', 'about', 'above', 'across', 'after', 'afterwards', 'again', 'against', 'all', 'almost', 'alone', 'along', 'already', 'also', 'although', 'always', 'am', 'among', 'amongst', 'amoungst', 'amount', 'an', 'and', 'another', 'any', 'anyhow', 'anyone', 'anything', 'anyway', 'anywhere', 'are', 'around', 'as', 'at', 'back', 'be', 'became', 'because', 'become', 'becomes', 'becoming', 'been', 'before', 'beforehand', 'behind', 'being', 'below', 'beside', 'besides', 'between', 'beyond', 'bill', 'both', 'bottom', 'but', 'by', 'call', 'can', 'cannot', 'cant', 'co', 'computer', 'con', 'could', 'couldnt', 'cry', 'de', 'describe', 'detail', 'do', 'done', 'down', 'due', 'during', 'each', 'eg', 'eight', 'either', 'eleven', 'else', 'elsewhere', 'empty', 'enough', 'etc', 'even', 'ever', 'every', 'everyone', 'everything', 'everywhere', 'except', 'few', 'fifteen', 'fify', 'fill', 'find', 'fire', 'first', 'five', 'for', 'former', 'formerly', 'forty', 'found', 'four', 'from', 'front', 'full', 'further', '

## Step 2: Text data preprocessing

Generate a function or functions to preprocess the 6 lists (i.e., 1. apple_pros, 2. apple_cons, 3. fb_pros, 4. fb_cons, 5. google_pros, and 6. google_cons) with the following purposes:
1. Lower case
2. Remove leading and ending spaces
2. Remove punctuations like @ : . , ?
3. Remove new line character \n and replace it with space
4. Convert all reviews to words

Explain how your functions work using Markdown cells

In [23]:
#create a function to format the sentence and create a list of words
def review_preprocess(reviews):
    formatted_list = []
    for sentence in reviews:
        formatted_word = sentence.lower(
            ).replace('@',''
            ).replace(':',''
            ).replace('.',''
            ).replace(',',''
            ).replace('?',''
            ).replace('\n', ' '
            ).split(
        )     
        formatted_list.extend(formatted_word)
    return formatted_list

Plug the 6 lists into your functions and get 6 proprocessed word lists

In [24]:
#save the formatted lists using the review_preprocess function to variables
apple_pros_formatted = review_preprocess(apple_pros)
apple_cons_formatted = review_preprocess(apple_cons)

fb_pros_formatted = review_preprocess(facebook_pros)
fb_cons_formatted = review_preprocess(facebook_cons)

google_pros_formatted = review_preprocess(google_pros)
google_cons_formatted = review_preprocess(google_cons)

Plaase print out the **number of words** in each word list

In [25]:
print('Number of words in Apple Pros: ' + str(len(apple_pros_formatted)))
print('Number of words in Apple Cons: ' + str(len(apple_cons_formatted)))

print('Number of words in Facebook Pros: ' + str(len(fb_pros_formatted)))
print('Number of words in Facebook Cons: ' + str(len(fb_cons_formatted)))

print('Number of words in Google Pros: ' + str(len(google_pros_formatted)))
print('Number of words in Google Cons: ' + str(len(google_cons_formatted)))

Number of words in Apple Pros: 5278
Number of words in Apple Cons: 10426
Number of words in Facebook Pros: 5845
Number of words in Facebook Cons: 7333
Number of words in Google Pros: 3896
Number of words in Google Cons: 8263


In [26]:
#store the number of words of each as a variable
apple_pros_total = len(apple_pros_formatted)
apple_cons_total = len(apple_cons_formatted)

fb_pros_total = len(fb_pros_formatted)
fb_cons_total = len(fb_cons_formatted)

google_pros_total = len(google_pros_formatted)
google_cons_total = len(google_cons_formatted)

## Step 3: Employee Satisfaction Score Calculation

### Goal: 
1. Calculate the **positive score** for each firm from their **pro reviews**
2. Calculate the **negative score** for each firm from their **con reviews**
3. Calculate **Employee Satisfaction Score** = positive score - negative score

### Analyze Apple, fb, and Google's pro review word lists

Generate a function or functions to get the

**1. positive word count**, 
**2. stop word count**, 
**3. total word count** 

from each firm's **pro** review word list.

In [27]:
#create a function to get the pro scores
#if the word is in the positive list, add it to the positive list count
#if the word is in the stop list, add it to the stop list count
def get_pro_scores(preprocessed_reviews):
    pos_word_count = 0
    stop_word_count = 0
    for word in preprocessed_reviews:
        if word in pos_list:
            pos_word_count += 1
        if word in stop_list:
            stop_word_count += 1
    return pos_word_count, stop_word_count, len(preprocessed_reviews)
#return the total positive word count, total stop word count, and the total word count
'''
    Generate a function or functions to get 
    1. positive word count
    2. stop word count
    3. total word count
    from each firm's pro review word list
    
    You can combine all stpes in one function or seperate them into multiple functions
    '''

"\n    Generate a function or functions to get \n    1. positive word count\n    2. stop word count\n    3. total word count\n    from each firm's pro review word list\n    \n    You can combine all stpes in one function or seperate them into multiple functions\n    "

1. Use the functions to calculate the **1. positive word count**, **2. stop word count**, **3. total word count**
2. Calculate each firm's **Positive score = Positive word count/ (Total word count - Stop word count)**

In [28]:
#run the get_pro_scores function and storing variables for Positive word count, Stop word count, Total word count
apple_pos_word_count, apple_stop_word_count, apple_total_pros_word_count = get_pro_scores(apple_pros_formatted)
fb_pos_word_count, fb_stop_word_count, fb_total_pros_word_count = get_pro_scores(fb_pros_formatted)
google_pos_word_count, google_stop_word_count, google_total_pros_word_count = get_pro_scores(google_pros_formatted)

In [29]:
#make a function to calculate the firm's positive score; takes the positive word count, stop word count, and total word count
def firm_pos_score(pos_word_count, stop_word_count, total_pro_word_count):
    pos_score = pos_word_count/(total_pro_word_count-stop_word_count)
    return pos_score

In [30]:
#run the firm_pos_score function and saving each firm as a variable
apple_firm_pos_score = firm_pos_score(apple_pos_word_count, apple_stop_word_count, apple_total_pros_word_count)
fb_firm_pos_score = firm_pos_score(fb_pos_word_count, fb_stop_word_count, fb_total_pros_word_count)
google_firm_pos_score = firm_pos_score(google_pos_word_count, google_stop_word_count, google_total_pros_word_count)

Print out Apple's 
1. Positive word count
2. Stop word count
3. Total word count 
4. Firm positive score

In [31]:
print('Apple\n'
    'Positive word count: ' + str(apple_pos_word_count) + '\n'
    'Stop word count: ' + str(apple_stop_word_count) + '\n'
    'Total word count: ' + str(apple_total_pros_word_count) + '\n'
    'Firm positive score: ' + str(apple_firm_pos_score) + '\n'
)

Apple
Positive word count: 1095
Stop word count: 1953
Total word count: 5278
Firm positive score: 0.3293233082706767



Print out fb's
1. Positive word count
2. Stop word count
3. Total word count 
4. Firm positive score

In [32]:
print('Facebook\n'
    'Positive word count: ' + str(fb_pos_word_count) + '\n'
    'Stop word count: ' + str(fb_stop_word_count) + '\n'
    'Total word count: ' + str(fb_total_pros_word_count) + '\n'
    'Firm positive score: ' + str(fb_firm_pos_score) + '\n'
)

Facebook
Positive word count: 841
Stop word count: 2409
Total word count: 5845
Firm positive score: 0.24476135040745053



Print out Google's 
1. Positive word count
2. Stop word count
3. Total word count 
4. Firm positive score

In [33]:
print('Google\n'
    'Positive word count: ' + str(google_pos_word_count) + '\n'
    'Stop word count: ' + str(google_stop_word_count) + '\n'
    'Total word count: ' + str(google_total_pros_word_count) + '\n'
    'Firm positive score: ' + str(google_firm_pos_score) + '\n'
)

Google
Positive word count: 907
Stop word count: 1354
Total word count: 3896
Firm positive score: 0.35680566483084186



### Analyze Apple, fb, and Google's con review word lists

Generate a function or functions to get the

**1. Negative word count**, 
**2. stop word count**, 
**3. total word count** 

from each firm's **con** review word list.

In [34]:
def get_con_scores(preprocessed_reviews):
    neg_word_count = 0
    stop_word_count = 0
    for word in preprocessed_reviews:
        if word in neg_list:
            neg_word_count += 1
        if word in stop_list:
            stop_word_count += 1
    return neg_word_count, stop_word_count, len(preprocessed_reviews) 

    '''
    Generate a function or functions to get 
    1. negative word count
    2. stop word count
    3. total word count
    from each firm's con review word list
    
    You can combine all stpes in one function or seperate them into multiple functions
    ''' 

1. Use the function to calculate the **1. Negative word count**, **2. stop word count**, **3. total word count**
2. Calculate each firm's **Negative score = Negative word count/ (Total word count - Stop word count)**

In [35]:
#store variables from the function above for each company
apple_neg_word_count, apple_stop_word_count, apple_total_neg_word_count = get_con_scores(apple_cons_formatted)
fb_neg_word_count, fb_stop_word_count, fb_total_neg_word_count = get_con_scores(fb_cons_formatted)
google_neg_word_count, google_stop_word_count, google_total_neg_word_count = get_con_scores(google_cons_formatted)


Print out Apple's 
1. Negative word count
2. Stop word count
3. Total word count 
4. Firm negative score

In [36]:
#make a function to calculate the firm negative score that takes 3 arguments
def firm_neg_score(neg_word_count, stop_word_count, total_word_count):
    neg_score = neg_word_count/(total_word_count-stop_word_count)
    return neg_score

In [37]:
#store the negative score as a variable for each firm
apple_firm_neg_score = firm_neg_score(apple_neg_word_count, apple_stop_word_count, apple_total_neg_word_count)
fb_firm_neg_score = firm_neg_score(fb_neg_word_count, fb_stop_word_count, fb_total_neg_word_count)
google_firm_neg_score = firm_neg_score(google_neg_word_count, google_stop_word_count, google_total_neg_word_count)

In [38]:
print('Apple\n'
    'Negative word count: ' + str(apple_neg_word_count) + '\n'
    'Stop word count: ' + str(apple_stop_word_count) + '\n'
    'Total word count: ' + str(apple_total_neg_word_count) + '\n'
    'Firm negative score: ' + str(apple_firm_neg_score) + '\n'
)

Apple
Negative word count: 328
Stop word count: 5055
Total word count: 10426
Firm negative score: 0.061068702290076333



Print out fb's 
1. Negative word count
2. Stop word count
3. Total word count 
4. Firm negative score

In [39]:
print('Facebook\n'
    'Negative word count: ' + str(fb_neg_word_count) + '\n'
    'Stop word count: ' + str(fb_stop_word_count) + '\n'
    'Total word count: ' + str(fb_total_neg_word_count) + '\n'
    'Firm negative score: ' + str(fb_firm_neg_score) + '\n'
)

Facebook
Negative word count: 264
Stop word count: 3544
Total word count: 7333
Firm negative score: 0.06967537608867776



Print out Google's 
1. Negative word count
2. Stop word count
3. Total word count 
4. Firm negative score

In [40]:
print('Google\n'
    'Negative word count: ' + str(google_neg_word_count) + '\n'
    'Stop word count: ' + str(google_stop_word_count) + '\n'
    'Total word count: ' + str(google_total_neg_word_count) + '\n'
    'Firm negative score: ' + str(google_firm_neg_score) + '\n'
)

Google
Negative word count: 265
Stop word count: 4071
Total word count: 8263
Firm negative score: 0.06321564885496184



### Calculate Apple, fb, and Google's employee satisfaction scores

Recall the Employee Satisfaction Score = Firm positive score - Firm negative score
<br>Print out each firm's Employee Satisfaction Score

In [41]:
#make a function to calculate the employee satisfaction score
def emp_sat_score(firm_pos_score, firm_neg_score):
    score = firm_pos_score - firm_neg_score
    return score

In [42]:
#run the above function and storing each score as a variable
apple_emp_sat_score = emp_sat_score(apple_firm_pos_score, apple_firm_neg_score)
fb_emp_sat_score = emp_sat_score(fb_firm_pos_score, fb_firm_neg_score)
google_emp_sat_score = emp_sat_score(google_firm_pos_score, google_firm_neg_score)

In [43]:
#print each score
print ('Apple Employee Satisfaction Score: ' + str(apple_emp_sat_score) + '\n')
print ('Facebook Employee Satisfaction Score: ' + str(fb_emp_sat_score) + '\n')
print ('Google Employee Satisfaction Score: ' + str(google_emp_sat_score) + '\n')

Apple Employee Satisfaction Score: 0.26825460598060036

Facebook Employee Satisfaction Score: 0.17508597431877276

Google Employee Satisfaction Score: 0.29359001597588



## Conclusion (Write your conclusion in the markdown cell below)

Which firm has better employee satisfaction during COVID-19 pandemic?

According to the analysis performed above, Google has the highest employee satisfaction score of 0.29 followed by Apple at 0.27, followed by Facebook at 0.18.

## Limitation

Please list at least one limitation of this analysis:

This analysis depends on the number of positive, negative, and stop words that, may in the context of the review, not necessarily indicate a more positive or negative review. For example, one review in cons for Facebook states, "I don't feel any cons so far," and would contribute towards a negative score as "cons" is on the negative list when the review is actually positive. 